src/ucs/profile/profile_defs.h

```diff
@@ -137,6 +161,54 @@ void ucs_profile_global_cleanup();
  */
 void ucs_profile_dump();
 
+
+/*
+ * Start a range trace on an arbitrary event in a potentially nested fashion.
+ * A range tracing can be started in a function and potentially end in an
+ * another function or a recursive invocation of the same function.
+ * A unique ID is returned to stop tracing the event.
+ *
+ * @param [in]     format      String name for the range.
+ *
+ * @return ID to be used to stop tracing a range
+ */
+uint64_t ucs_profile_range_start(const char *format, ...);
+
+
+/*
+ * Stop range trace.
+ *
+ * @param [in]     id          id that was returned from range start.
+ *
+ */
+void ucs_profile_range_stop(uint64_t id);
+
+
+/*
+ * Add a marker on trace profiles.
+ *
+ * @param [in]     format      String name for the marker.
+ *
+ */
+void ucs_profile_range_add_marker(const char *format, ...);
+
+
+/*
+ * Start a range trace in a non-nested fashion. Range tracing must start and end
+ * in the same function.
+ *
+ * @param [in]     format      String name for the marker.
+ *
+ */
+void ucs_profile_range_push(const char *format, ...);
```

Comment:
is it disabled in release mode?

Reply:
The implementation resolves to a no-op. Is that what you mean by disable here? 

The call to ucs_profile_range* are still made in the UCP/UCT transfer paths so this does add function call overheads unlike the UCS_PROFILE_* macros.